api/layer/tagging.go

```diff
@@ -3,48 +3,54 @@ package layer
 import (
 	"context"
 	errorsStd "errors"
+	"strings"
 
 	"github.com/nspcc-dev/neofs-s3-gw/api/data"
 	"github.com/nspcc-dev/neofs-s3-gw/api/errors"
 	cid "github.com/nspcc-dev/neofs-sdk-go/container/id"
 	"go.uber.org/zap"
 )
 
-func (n *layer) GetObjectTagging(ctx context.Context, p *ObjectVersion) (map[string]string, error) {
+func (n *layer) GetObjectTagging(ctx context.Context, p *ObjectVersion) (string, map[string]string, error) {
 	var (
 		err  error
 		tags map[string]string
 	)
-	tags = n.systemCache.GetTagging(objectTaggingCacheKey(p))
+
+	key := objectTaggingCacheKey(p)
+	tags = n.systemCache.GetTagging(key)
 	if tags != nil {
-		return tags, nil
+		split := strings.Split(key, ".")
+		return split[len(split)-1], tags, nil
```

Comment:
If `tags != nil` then version is stored in `p.VersionID`, isn't it, thus string splitting is not required, isn't it?